grammar/haxe.bnf

```diff
@@ -254,11 +254,12 @@ private topLevelDeclaration ::= classDeclaration
 
 privateKeyWord ::= 'private'
 externOrPrivate ::= 'extern' | privateKeyWord
+private externAndMaybePrivate ::= privateKeyWord? 'extern' privateKeyWord?
 
 typedefDeclaration ::= macroClassList? externOrPrivate? 'typedef' componentName genericParam? '=' functionTypeWrapper ';'?
 {pin=5 mixin="com.intellij.plugins.haxe.lang.psi.impl.AbstractHaxeTypeDefImpl" implements="com.intellij.plugins.haxe.lang.psi.HaxeClass"}
 
-externClassDeclaration ::= macroClassList? externOrPrivate* ('class' | 'interface') componentName genericParam? inheritList? '{' externClassDeclarationBody '}'
+externClassDeclaration ::= macroClassList? externAndMaybePrivate ('class' | 'interface') componentName genericParam? inheritList? '{' externClassDeclarationBody '}'
```

Comment:
Yeah, the main purpose of this pull was to prevent parsing regular interfaces as extern classes.

>Don't you want to drop the 'interface' from the OR clause here?

I don't get what do you mean by this

>Do you mean allow extern modifier for interfaceDeclaration(as it used before this pull request), and remove interface from externalClassDeclaration? (additionally add check for AbstractHaxePsiClass.java/isExtern function for external interface?)

I think this is alternative solution, also rules for external classes and external interfaces may be a lil bit different

I tested it in try.haxe.org, 

extern class allows methods to have body(empty)
extern interface doesn't(so it might be closed to regular interface parsing rules)

Maybe it's complicated